### PR TITLE
Avoid `T.must`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/formula.rb [hunk 1]
```diff
@@ -540,8 +540,8 @@ def oldname
   # Old names for the formula.
   sig { returns(T::Array[String]) }
   def oldnames
-    @oldnames ||= if tap
-      T.must(tap).formula_oldnames.fetch(name, [])
+    @oldnames ||= if (tap = self.tap)
+      tap.formula_oldnames.fetch(name, [])
     else
       []
     end
```
Comment (dduugg) [inline Library/Homebrew/formula.rb:547]: wdyt of:
```suggestion
    @oldnames ||= tap&.formula_oldnames&.fetch(name, []) || []
```
(Or maybe `&.[](name)` if you want to code golf it)
?

Comment (reitermarkus) [reply Library/Homebrew/formula.rb:547]: If I really wanted to golf it:

```suggestion
    @oldnames ||= [*tap&.formula_oldnames&.[](name)]
```

😂

Comment (reitermarkus) [reply Library/Homebrew/formula.rb:547]:
```suggestion
    @oldnames ||= (tap&.formula_oldnames || {}).fetch(name, [])
```

maybe?

Comment (reitermarkus) [reply Library/Homebrew/formula.rb:547]:
```suggestion
    @oldnames ||= tap&.formula_oldnames&.dig(name) || []
```

is probably the most readable I can come up with.

Comment (apainintheneck) [reply Library/Homebrew/formula.rb:547]: I'm pretty sure the single method dig will unfortunately not pass Rubocop by default. We could disable that rule though. I like the last one best but I don't really care that much.

File: Library/Homebrew/formula.rb [hunk 2]
```diff
@@ -540,23 +540,13 @@ def oldname
...
   sig { returns(T::Array[String]) }
   def aliases
-    @aliases ||= if tap
-      T.must(tap).alias_reverse_table[full_name].to_a.map do |a|
-        a.split("/").last
-      end
-    else
-      []
-    end
+    @aliases ||= tap&.alias_reverse_table&.dig(full_name)&.map { |a| a.split("/").last } || []
```
Comment (dduugg) [inline Library/Homebrew/formula.rb:549]:
```suggestion
    @aliases ||= tap&.alias_reverse_table&.dig(full_name)&.map { _1.split("/").last } || []
```

**Review by dduugg — COMMENTED** [review]
I think I don't feel strongly either way, my general sense of readability is:

nil forwarding > T.must > assignment in conditionals

but this feels a bit like a wash to me overall

**Review by apainintheneck — APPROVED** [review]
This looks good. Does this pattern show up anywhere else in the codebase? We could probably just replace those usages with this one if more exist.

Also, how militant should we be about removing these extra runtime Sorbet type assertions in general?

**Comment by reitermarkus** [comment]
I just stumbled over these while refactoring some other alias-related code, so it made sense to also refactor this, but I don't think we need to proactively do anything about this pattern and assertions in general. Of course feel free to take a look, I'd assume this pattern is used in a couple of other places.

File: Library/Homebrew/formula.rb [hunk 3]
```diff
@@ -540,23 +540,13 @@ def oldname
   # Old names for the formula.
   sig { returns(T::Array[String]) }
   def oldnames
-    @oldnames ||= if tap
-      T.must(tap).formula_oldnames.fetch(name, [])
-    else
-      []
-    end
+    @oldnames ||= tap&.formula_oldnames&.dig(name) || []
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula.rb:543]: Do we expect `tap` to ever be not `nil` but `formula_oldnames` is nil`?

If not, the existing code but just without the `T.must` feels nicer to me than repeatedly chaining `&`

**Pull request merged**